### ISSUE TITLE
Ghost profile(s)?

### ISSUE THREAD

**Issue by cybeardjm** [Automattic/wordpress-activitypub#457]
### Quick summary

OK, Not sure if the problem comes from Mastodon/FireFish, WP/AP 1.0 or ActivityPub... Let me explain below.

TL;DR: when recalling via search an account from a WP/AP blog, that was just unfollowed, all the posts from the previous connection still appear.
Expected (IMHO): a clean fetch from the source, with no posts and latest description, as happens when a distant account is first followed.
The only data that seems to be updated is the number of followers...

### Steps to reproduce

I've been following my 2 WP/AP blogs since end of last year (they existed before, but I moved to a new instance). Later I created a Calckey/FireFish account for testing purpose, and followed my 2 blogs there too. I then published many posts that I can see under each account on both platforms.

As I've been testing potential conflicts between the AIOWPS plugin and AP 1.0, I unfollowed both blogs from Mastodon & Firefish.

Now, if I try to follow these accounts again, through search, I can find them and previsualize them.

But it seems a "cached" version appears in both Mastodon & FireFish, with all the previous posts and the previous description (even if it's changed in the meantime).

PS: for an unknown reason, for one of my blogs, the new account's description appeared, but not for the second one. All "old" posts were still visible in both cases.

### What you expected to happen

After the account has been unfollowed, visualizing it again should fetch a "clean" profile... what happens when an account is followed for the first time, as "old" posts don't appear, only the ones published afterwards.

Like a kind of "cache flush" somewhere in the relationship between the 2 accounts...

### What actually happened

All the posts published when the account was previously followed still appear even though it's not yet followed again.

### Impact

One

### Available workarounds?

No but the platform is still usable

**Comment by timnolte** [comment]
I'm pretty sure this is simply a Mastodon, if that is what you are testing on, instance caching issue. What content shows up when viewing a profile when you aren't following someone is all in the hands of the instance/ActivityPub platform you are using. There are know GitHub issues regarding backfilling user profiles when viewing in order to see all public content. I suspect since you were already following the blog the instance already had all of the content in it's cache so it would naturally display that.
